aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -0,0 +1,295 @@
+/// This module provides the foundation for (collectible) Tokens often called NFTs
+module AptosFramework::Token {
+    use Std::ASCII;
+    use Std::GUID::{Self, ID};
+    use Std::Option::{Self, Option};
+    use Std::Signer;
+    use AptosFramework::Table::{Self, Table};
+
+    // Error map
+    const EINSUFFICIENT_BALANCE: u64 = 0;
+    const EMISSING_CLAIMED_TOKEN: u64 = 1;
+
+    // A creator may publish multiple collections
+    struct Collections<TokenType: copy + drop + store> has key {
+        collections: Table<ID, Collection<TokenType>>,
+    }
+
+    fun initialize_collections<TokenType: copy + drop + store>(account: &signer) {
+        move_to(
+            account,
+            Collections {
+                collections: Table::create<ID, Collection<TokenType>>(),
+            },
+        )
+    }
+
+    // The source of Tokens, their collection!
+    struct Collection<TokenType: copy + drop + store> has store {
+        // Keep track of all Tokens, even if their balance is 0.
```

Comment:
keep track of all metadata?

Reply:
will update in follow up as we change the api around per conversation